app/src/org/commcare/activities/QueryRequestActivity.java

```diff
@@ -0,0 +1,335 @@
+        try {
+            datum = sessionWrapper.getSession().getNeededDatum();
+        } catch (NullPointerException e) {
+            // tried loading session info when it wasn't there
+            return null;
+        }
+        if (datum instanceof RemoteQueryDatum) {
+            return new RemoteQuerySessionManager((RemoteQueryDatum)datum,
+                    sessionWrapper.getEvaluationContext());
+        } else {
+            return null;
+        }
+    }
+
+    private void setupUI() {
+        buildPromptUI();
+
+        queryButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                ViewUtil.hideVirtualKeyboard(QueryRequestActivity.this);
+                answerPrompts();
+                makeQueryRequest();
+            }
+        });
+
+        if (inErrorState) {
+            enterErrorState();
+        }
+    }
+
+    private void buildPromptUI() {
+        LinearLayout promptsLayout = (LinearLayout)findViewById(R.id.query_prompts);
+        Hashtable<String, DisplayUnit> userInputDisplays =
+                remoteQuerySessionManager.getNeededUserInputDisplays();
+        int promptCount = 1;
+        for (Map.Entry<String, DisplayUnit> displayEntry : userInputDisplays.entrySet()) {
+            boolean isLastPrompt = promptCount++ == userInputDisplays.size();
+            buildPromptEntry(promptsLayout, displayEntry.getKey(),
+                    displayEntry.getValue(), isLastPrompt);
+        }
+    }
+
+    private void buildPromptEntry(LinearLayout promptsLayout, String promptId,
+                                  DisplayUnit displayUnit, boolean isLastPrompt) {
+        Hashtable<String, String> userAnswers =
+                remoteQuerySessionManager.getUserAnswers();
+        promptsLayout.addView(createPromptMedia(displayUnit));
+
+        EditText promptEditText = new EditText(this);
+        if (userAnswers.containsKey(promptId)) {
+            promptEditText.setText(userAnswers.get(promptId));
+        }
+        promptEditText.setSingleLine();
+        promptEditText.setBackgroundResource(R.drawable.login_edit_text);
+        if (isLastPrompt) {
+            promptEditText.setImeOptions(EditorInfo.IME_ACTION_DONE);
+        } else {
+            // replace 'done' on keyboard with 'next'
+            promptEditText.setImeOptions(EditorInfo.IME_ACTION_NEXT);
+        }
+        promptsLayout.addView(promptEditText);
+        promptsBoxes.put(promptId, promptEditText);
+    }
+
+    private MediaLayout createPromptMedia(DisplayUnit display) {
+        DisplayData displayData = display.evaluate();
+        String promptText =
+                Localizer.processArguments(displayData.getName(), new String[]{""}).trim();
+        TextView text = new TextView(getApplicationContext());
+        text.setText(promptText);
+
+        int padding = (int)getResources().getDimension(R.dimen.help_text_padding);
+        text.setPadding(0, 0, 0, 7);
```

Comment:
Agreed on the migrate. In the short term, you should be able to add the values you want to use to `dimens.xml` and then reference those here using `getResources().getDimension(R.dimen.xxx)`